wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -436,6 +443,21 @@ private void startAsync(
       throw new ConflictException(e.getMessage(), e);
     }
 
+    // handle the situation where a workspace created by a previous Che version doesn't have a
+    // namespace stored for it. In this case, we just store the default namespace for it.
+    if (isNullOrEmpty(
```

Comment:
Would it be possible to fix workspace object as soon as possible in the chain, so we would avoid `if`s like this in the code later?

Reply:
This is the soonest we can do it in a normal flow. We could theoretically do this during workspace recovery, but that would be a much larger change, because workspace recovery currently only reconciles running workspaces.